diskmaker/controllers/lv/reconcile_test.go

```diff
@@ -318,34 +347,51 @@ func getDeiveIDs() []string {
 }
 
 func createTmpDir(t *testing.T, dir, prefix string) string {
-	tmpDir, err := ioutil.TempDir(dir, prefix)
+	tmpDir, err := os.MkdirTemp(dir, prefix)
 	if err != nil {
 		t.Fatalf("error creating temp directory : %v", err)
 	}
 	return tmpDir
 }
 
-func createTmpFile(t *testing.T, dir, pattern string) *os.File {
-	tmpFile, err := ioutil.TempFile(dir, pattern)
+func createTmpFile(t *testing.T, dir, pattern string, size int64) *os.File {
+	tmpFile, err := os.CreateTemp(dir, pattern)
 	if err != nil {
 		t.Fatalf("error creating tmp file: %v", err)
 	}
+	err = tmpFile.Truncate(size)
+	if err != nil {
+		t.Fatalf("error truncating tmp file: %v", err)
+	}
 	return tmpFile
 }
 
 func hasFile(t *testing.T, dir, file string) bool {
-	files, err := ioutil.ReadDir(dir)
+	dentries, err := os.ReadDir(dir)
 	if err != nil {
 		t.Fatalf("error reading directory %s : %v", dir, err)
 	}
-	for _, f := range files {
-		if strings.Contains(f.Name(), file) {
+	for _, d := range dentries {
+		if strings.Contains(d.Name(), file) {
 			return true
 		}
 	}
 	return false
 }
 
+func formatAsExt4(t *testing.T, fname string) {
```

Comment:
I believe this should move under [VolumeUtil interface](https://github.com/openshift/local-storage-operator/blob/4e05a300cc0a5ce7f2755b4908332125bcf5074f/vendor/sigs.k8s.io/sig-storage-local-static-provisioner/pkg/util/volume_util.go#L24) and since this is Linux specific the implementation should be in [this](https://github.com/openshift/local-storage-operator/blob/4e05a300cc0a5ce7f2755b4908332125bcf5074f/vendor/sigs.k8s.io/sig-storage-local-static-provisioner/pkg/util/volume_util_linux.go) file.

Then the implementation in `volume_util_unsupported.go` should get these new methods (`formatAsExt4` and `hasExt4`) as well to throw unsupported error when your test runs on different platform.

Reply:
@RomanBednar , [VolumeUtil interface](https://github.com/openshift/local-storage-operator/blob/4e05a300cc0a5ce7f2755b4908332125bcf5074f/vendor/sigs.k8s.io/sig-storage-local-static-provisioner/pkg/util/volume_util.go#L24) comes from upstream `sig-storage-local-static-provisioner` , and it doesn't need `formatAsExt4()`.

I don't think we can (or have to) pollute upstream repo with some code that is required only for LSO, especially only for LSO tests. Theoretically, we could implement similar stuff (`volume_util*.go`) somewhere downstream, but before doing that -- Do we support LSO on any different (than linux) platform? If not, why we're intersted about running tests on different platforms?

Reply:
@mpatlasov Ah, it's for testing only. So you're right, it should not be in `sig-storage-local-static-provisioner`. 

I don't think we support anything else than Linux in LSO, but the unit test itself will fail weird if not run on Linux:

```
=== RUN   TestWipeDeviceWhenCreateSymLinkByDeviceName
=== RUN   TestWipeDeviceWhenCreateSymLinkByDeviceName/forceWipeDevicesAndDestroyAllData_is_False
    reconcile_test.go:386: error formatting file: /var/folders/98/w1dp6fq94k53hgy7ml6xqz8h0000gn/T/diskName4038852681; command error: exec: "mkfs.ext4": executable file not found in $PATH
=== RUN   TestWipeDeviceWhenCreateSymLinkByDeviceName/forceWipeDevicesAndDestroyAllData_is_True
    reconcile_test.go:386: error formatting file: /var/folders/98/w1dp6fq94k53hgy7ml6xqz8h0000gn/T/diskName3925205785; command error: exec: "mkfs.ext4": executable file not found in $PATH
--- FAIL: TestWipeDeviceWhenCreateSymLinkByDeviceName (0.00s)
    --- FAIL: TestWipeDeviceWhenCreateSymLinkByDeviceName/forceWipeDevicesAndDestroyAllData_is_False (0.00s)
```

If we do something similar to `VolumeUtil` like [this](https://github.com/RomanBednar/local-storage-operator/commit/227a51d407850441d622d6e2f8b691216691f0cd) we can directly say it's unsupported at build:

```
--- FAIL: TestWipeDeviceWhenCreateSymLinkByDeviceName (0.00s)
    --- FAIL: TestWipeDeviceWhenCreateSymLinkByDeviceName/forceWipeDevicesAndDestroyAllData_is_False (0.00s)
panic: FormatAsExt4 is unsupported in this build [recovered]
	panic: FormatAsExt4 is unsupported in this build
```

The ideal case I believe would be to change `make test` to run tests in a container, but that's out of scope of this PR.

Maybe the platform specific builds are not worth the hassle in this case, feel free to keep it as it is for now.

Reply:
@RomanBednar thank you for example implementation of `test-framework/testutils/volume_util*.go` -- that's what I meant saying we could implement similar stuff (volume_util*.go) somewhere downstream. I added this code to [commit `ff29c85`](https://github.com/openshift/local-storage-operator/pull/431/commits/ff29c851192776f5c89c0f807f2f8a55cfebb810), so we have "unsupported" message at build time now.